### PR TITLE
Adding trivy action

### PR DESCRIPTION
**Pull request by arkadiuszbeer**

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -173,3 +173,14 @@ jobs:
         run: |
           rm -rf /tmp/.buildx-cache
           mv /tmp/.buildx-cache-new /tmp/.buildx-cache
+
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@master
```
Comment (walkowif) [inline .github/workflows/deploy.yaml:178]: Just to consider: maybe using specific version of the action would make sense?

**Review by walkowif — APPROVED** [review]

File: .github/workflows/deploy.yaml (hunk 1 above)
Comment (dinakar29) [inline .github/workflows/deploy.yaml:178]:
```suggestion
        uses: aquasecurity/trivy-action@v0.4.1
```

File: .github/workflows/scheduled.yaml [hunk 2]
```diff
@@ -39,3 +39,18 @@ jobs:
               "tag": "",
               "tag_latest": "true"
             }
+      - name: Build an image from Dockerfile
+        run: |
+          docker build -t ${{ matrix.image.distro }}:${{ github.sha }} .
```
Comment (dinakar29) [inline .github/workflows/scheduled.yaml:44]: Nope. Please do not do this. This is extremely wasteful.

Simply fetch the Docker image from the registry instead.

File: .github/workflows/scheduled.yaml [hunk 3]
```diff
@@ -39,3 +39,18 @@ jobs:
...
+
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@master
```
Comment (dinakar29) [inline .github/workflows/scheduled.yaml:47]: Change the version

File: .github/workflows/trivy.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Build an image from Dockerfile
+        run: |
+          docker build -t docker.io/my-organization/my-app:${{ github.sha }} .
+
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@master
```
Comment (dinakar29) [inline .github/workflows/trivy.yaml:24]: Change version

File: .github/workflows/trivy.yaml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        with:
+          image-ref: 'docker.io/my-organization/my-app:${{ github.sha }}'
+          format: 'table'
+          exit-code: '1'
+          ignore-unfixed: true
+          vuln-type: 'os,library'
+          severity: 'CRITICAL,HIGH'
```
Comment (dinakar29) [inline .github/workflows/trivy.yaml:31]: Missing newline

File: .github/workflows/trivy.yaml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        uses: actions/checkout@v2
+
+      - name: Build an image from Dockerfile
+        run: |
+          docker build -t docker.io/my-organization/my-app:${{ github.sha }} .
+
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@master
+        with:
+          image-ref: 'docker.io/my-organization/my-app:${{ github.sha }}'
```
Comment (dinakar29) [inline .github/workflows/trivy.yaml:26]: Copy-paste error

**Review by dinakar29 — CHANGES_REQUESTED** [review]
Please see inline comments

File: .github/workflows/deploy.yaml (hunk 1 above)
Comment (arkadiuszbeer) [reply .github/workflows/deploy.yaml]: done

File: .github/workflows/scheduled.yaml (hunk 3 above)
Comment (arkadiuszbeer) [reply .github/workflows/scheduled.yaml]: done

File: .github/workflows/trivy.yaml (hunk 4 above)
Comment (arkadiuszbeer) [reply .github/workflows/trivy.yaml]: done

File: .github/workflows/trivy.yaml (hunk 6 above)
Comment (arkadiuszbeer) [reply .github/workflows/trivy.yaml]: fixed

File: .github/workflows/trivy.yaml (hunk 5 above)
Comment (arkadiuszbeer) [reply .github/workflows/trivy.yaml]: done

File: .github/workflows/trivy.yaml [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  build:
+    name: Build
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Build an image from Dockerfile
+        run: |
+          docker build -t docker.io/my-organization/my-app:${{ github.sha }} .
```
[the remark this answers is not in the document]
Comment (arkadiuszbeer) [reply .github/workflows/trivy.yaml]: Trying to

File: .github/workflows/scheduled.yaml (hunk 2 above)
Comment (arkadiuszbeer) [reply .github/workflows/scheduled.yaml]: done

File: .github/workflows/scheduled.yaml [hunk 8]
```diff
@@ -39,3 +39,22 @@ jobs:
...
+
+      - name: Log in to the Container registry 🗝
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@7b7aa264d83dc58691451798b4d117d53d21edfe
```
Comment (dinakar29) [inline .github/workflows/scheduled.yaml:51]:
```suggestion
        uses: aquasecurity/trivy-action@master
```

Comment (dinakar29) [reply .github/workflows/scheduled.yaml:51]: Change to `master` everywhere to get the latest security patches, if and when Aqua makes them available

File: .github/workflows/scheduled.yaml [hunk 9]
```diff
@@ -39,3 +39,22 @@ jobs:
...
+        with:
+          image-ref: 'ghcr.io/insightsengineering/rstudio_4.1.2_bioc_3.14:latest'
```
Comment (dinakar29) [inline .github/workflows/scheduled.yaml:53]: This ought to come from a matrix.

File: .github/workflows/trivy.yaml [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+env:
+  REGISTRY: ghcr.io
+
+on:
+  schedule:
+    - cron: '44 19 7,21 * *'
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
```
Comment (dinakar29) [inline .github/workflows/trivy.yaml:12]: Not needed

File: .github/workflows/scheduled.yaml [hunk 11]
```diff
@@ -39,3 +39,22 @@ jobs:
...
+          format: 'sarif'
+          output: 'trivy-results.sarif'
+
+      - name: Upload Trivy scan results to GitHub Security tab
+        uses: github/codeql-action/upload-sarif@v2
+        with:
+          sarif_file: 'trivy-results.sarif'
```
Comment (dinakar29) [inline .github/workflows/scheduled.yaml:60]: I'd say remove the workflow from this schedule and keep it in `trivy.yaml`

File: .github/workflows/trivy.yaml [hunk 12]
```diff
@@ -0,0 +1,68 @@
+name: trivy  ⏲
```
Comment (dinakar29) [inline .github/workflows/trivy.yaml:1]: Call this `Security Scan`. Also rename the workflow to `security-scan.yaml`

File: .github/workflows/trivy.yaml [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@7b7aa264d83dc58691451798b4d117d53d21edfe
+        with:
+          image-ref: 'ghcr.io/insightsengineering/rstudio_4.1.2_bioc_3.14:latest'
```
Comment (dinakar29) [inline .github/workflows/trivy.yaml:44]: Make this a matrix, to match the one in `scheduled.yaml`

File: .github/workflows/trivy.yaml [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+            distro: rstudio
+    steps:
+      - name: Checkout repository 💳
+        uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+        id: buildx
+        with:
+          install: true
```
Comment (dinakar29) [inline .github/workflows/trivy.yaml:32]: These are not needed

File: .github/workflows/trivy.yaml [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+          sarif_file: 'trivy-results.sarif'
+
+      - name: Run Trivy vulnerability scanner in repo mode
+        uses: aquasecurity/trivy-action@7b7aa264d83dc58691451798b4d117d53d21edfe
+        with:
+          scan-type: 'fs'
+          ignore-unfixed: true
+          #format: 'sarif'
+          format: 'table'
+          severity: 'CRITICAL,HIGH'
```
Comment (dinakar29) [inline .github/workflows/trivy.yaml:66]: Do not run in repo mode

**Review by dinakar29 — CHANGES_REQUESTED** [review]

File: .github/workflows/scheduled.yaml (hunk 8 above)
Comment (arkadiuszbeer) [reply .github/workflows/scheduled.yaml:51]: done

File: .github/workflows/trivy.yaml (hunk 13 above)
Comment (arkadiuszbeer) [reply .github/workflows/trivy.yaml:44]: Using only one `rstudio-local`

File: .github/workflows/trivy.yaml (hunk 14 above)
Comment (arkadiuszbeer) [reply .github/workflows/trivy.yaml:32]: done

File: .github/workflows/scheduled.yaml (hunk 9 above)
Comment (arkadiuszbeer) [reply .github/workflows/scheduled.yaml]: done

File: .github/workflows/trivy.yaml (hunk 12 above)
Comment (arkadiuszbeer) [reply .github/workflows/trivy.yaml]: done

File: .github/workflows/trivy.yaml (hunk 10 above)
Comment (arkadiuszbeer) [reply .github/workflows/trivy.yaml]: I would leave `workflow_dispatch`

File: .github/workflows/security-scan.yaml [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+
+on:
+  schedule:
+    - cron: '44 21 7,21 * *'
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-18.04
```
Comment (dinakar29) [inline .github/workflows/security-scan.yaml:14]:
```suggestion
    runs-on: ubuntu-latest
```

File: .github/workflows/security-scan.yaml [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+          output: 'trivy-results.sarif'
+
+      - name: Upload Trivy scan results to GitHub Security tab ⛽
+        uses: github/codeql-action/upload-sarif@v2
+        if: always()
+        with:
+          sarif_file: 'trivy-results.sarif'
+
+
+
```
Comment (dinakar29) [inline .github/workflows/security-scan.yaml:50]:
```suggestion
```

**Review by dinakar29 — APPROVED** [review]

**Pull request merged**